### ISSUE TITLE
Support for EventBridge / Custom Event Buses

### ISSUE THREAD

**Issue by brandonstevens** [awslabs/serverless-application-model#1067]
[Amazon EventBridge](https://aws.amazon.com/about-aws/whats-new/2019/07/introducing-amazon-eventbridge/) was announced. One aspect of the announcement was added support for custom event buses. Custom buses provide a clean way to isolate events for custom applications. We should be able to specific a custom bus as the source for triggering Lambda functions.

Example template:

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:

  SlackNotify:
    Type: AWS::Serverless::Function
    Properties:
      Handler: main.handler
      CodeUri: app/
      Runtime: python3.7
      Timeout: 5
      Events:
        AnyRequest:
          Type: CloudWatchEvent
          Properties:
            Bus: example-custom-bus
            Pattern:
              source:
                - example.test
```

**Comment by jlhood** [comment]
@brandonstevens Thanks for opening this issue. Totally agree this will be a great addition to SAM. However, we're blocked from adding this until Amazon EventBridge adds CloudFormation support. I don't have a confirmed date, but I've heard in public talks that it's "coming soon." I do see an issue for it on CloudFormation's new [public coverage roadmap](https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/44), so that helps.

According to that issue, it will just be a new property added to the existing `AWS::Events::Rule` resource type. Assuming that's true, this should be fairly trivial to add once the CloudFormation support is there.

**Comment by zbintliff** [comment]
This is supported in CFN now!
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Events.html

**Comment by jlhood** [comment]
@zbintliff Yes it is! The change in SAM to support this should be really simple if someone from the community wants to contribute it. Update the [CloudWatchEvent source](https://github.com/awslabs/serverless-application-model/blob/develop/samtranslator/model/eventsources/push.py#L141) to take an `EventBusName` property and pass it through to the underlying EventsRule object.

**Comment by zbintliff** [comment]
I'll take it. I _think_ i'm mostly there but tests are failing and i can't quite parse why

**Comment by zbintliff** [comment]
Hey @jlhood  just wondering what the release schedule is for this application?

**Comment by ShreyaGangishetty** [comment]
@zbintliff We're working on making our release process more transparent to external customers, but we're not quite where we want to be yet. For now, you can see when we cut [release](https://github.com/awslabs/serverless-application-model/branches/active) branches, which can give you some indication of when we begin working on a specific release. Once a release tracker is created you can track its progress on the [release board](https://github.com/awslabs/serverless-application-model/projects/3) #

**Comment by zbintliff** [comment]
Hi Shreya, I took a look at the links and can't really see what will ultimately be included. Do we know if #1185 will make it on #1212 (this is only for our own planning as right now we have to script the creation of these rules)

**Comment by zbintliff** [comment]
@jlhood can you double check me? Will my change make sure that the permission on the lambda will allow the CW Rule  on the EventBridge external bus to trigger the lambda?

**Comment by jlhood** [comment]
@zbintliff I believe so. If you look at the output file from the unit test you modified, you can see the permission enabled the `events.amazonaws.com` service principal (EventBridge service) to invoke the Lambda function as long as the source is the `AWS::Events::Rule` resource ARN:

https://github.com/awslabs/serverless-application-model/blob/440790a8c2278fc16acbe34c83183668c5f40c38/tests/translator/output/cloudwatchevent.json#L27-L42

The `AWS::Events::Rule` resource is what is specifying the event bus name. The Lambda permission appears to be between the Rule ARN and the Lambda function ARN. It doesn't appear any special extra permission is required based on which event bus the Rule is configured against.

https://github.com/awslabs/serverless-application-model/blob/440790a8c2278fc16acbe34c83183668c5f40c38/tests/translator/output/cloudwatchevent.json#L83-L106

If you want to doublecheck, you can create a native CloudFormation template based off of this output file and verify it works as expected.

**Comment by zbintliff** [comment]
Awesome, thanks

**Comment by ShreyaGangishetty** [comment]
Closing this issue as this feature is released in [v1.17.0](https://github.com/awslabs/serverless-application-model/releases/tag/v1.17.0)
